runtime/moonbase/tests/integration_test.rs

```diff
@@ -547,7 +548,8 @@ fn join_candidates_via_precompile() {
 
 			// Construct the call data (selector, amount)
 			let mut call_data = Vec::<u8>::from([0u8; 68]);
-			call_data[0..4].copy_from_slice(&hex_literal::hex!("ad76ed5a"));
+			call_data[0..4]
+				.copy_from_slice(&Keccak256::digest(b"join_candidates(uint256,uint256)")[0..4]);
```

Comment:
Good test to verify the manually calculated function selectors. Why not use this where they are set so the code is more readable?

Reply:
It's probably cheaper to use hardcoded bytes if they don't ever change than to recalculate them by hashing every time the precompile needs to be accessed.

Reply:
I've been wanting to play with this, so I did this morning. I tried creating a `const fn`, which could be used at compile-time and not called at runtime:

```rust
const fn calculate_fn_selector(sig: &str) -> [u8; 4] {
    Keccak256::digest(b"join_candidates(uint256,uint256)")[0..4].try_into().expect("Wrong size")
}
```

This fails to compile because `Keccak256::digest()` is not itself a `const fn`. I don't see a way around this without some sacrifice at runtime...